packages/core/test/behaviors.test.ts

```diff
@@ -82,6 +85,34 @@ describe('promise behavior (fromPromise)', () => {
 
     expect(called).toBe(false);
   });
+
+  it('should persist a promise', (done) => {
+    const promiseBehavior = fromPromise(
+      () =>
+        new Promise<number>((res) => {
+          res(42);
+        })
+    );
+
+    const actor = interpret(promiseBehavior);
+    actor.start();
+
+    setTimeout(() => {
+      const resolvedPersistedState = actor.getPersistedState();
+
+      expect(resolvedPersistedState).toEqual(
+        expect.objectContaining({
+          data: 42
+        })
+      );
+
+      const restoredActor = interpret(promiseBehavior, {
+        state: resolvedPersistedState
+      }).start();
+      expect(restoredActor.getSnapshot()).toBe(42);
+      done();
+    }, 5);
+  });
```

Comment:
I think that this is an improved version of this test and it doesn't pass:
```suggestion
  it('should persist a promise', (done) => {
    let createdPromises = 0;
    const promiseBehavior = fromPromise(() => {
      createdPromises++;
      return Promise.resolve(createdPromises);
    });

    const actor = interpret(promiseBehavior);
    actor.start();

    setTimeout(() => {
      const resolvedPersistedState = actor.getPersistedState();

      expect(resolvedPersistedState).toEqual(
        expect.objectContaining({
          data: 1
        })
      );

      const restoredActor = interpret(promiseBehavior, {
        state: resolvedPersistedState
      }).start();
      expect(restoredActor.getSnapshot()).toBe(1);
      expect(createdPromises).toBe(1);
      done();
    }, 5);
  });
```

Reply:
Additionally - this test prints this in the console now:
```
    Warning: Event "$$xstate.resolve" was sent to stopped actor "x:1 (x:1)". This actor has already reached its final state, and will not transition.
    Event: {"type":"$$xstate.resolve","data":42}
```

Reply:
8c9ae224dffa7a14c5f1e74c1a7ac9bb873fd504